buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1373,12 +1377,17 @@ protected final void checkIndex(int index, int fieldLength) {
     }
 
     final void checkIndex0(int index, int fieldLength) {
-        if (isOutOfBounds(index, fieldLength, capacity())) {
-            throw new IndexOutOfBoundsException(String.format(
-                    "index: %d, length: %d (expected: range(0, %d))", index, fieldLength, capacity()));
+        int capacity = capacity();
```

Comment:
I understand the rational, and in theory it seems reasonable, but this is not required for the new `UTF8 decoding` feature. We try to keep changes separate as they are easier to verify, easier to port across branches, and if something goes wrong (not saying it will) easier to isolate issues.

In this other PR you will need to demonstrate the difference in byte code size between `checkIndex0` before and after, and also provide some demonstration (e.g using `-XX:+PrintInlining`) that the method is now more likely to be inlined inlined to support the claims (which seem reasonable).